minifi-api/src/main/java/org/apache/nifi/minifi/configuration/ConfigurationChangeNotifier.java

```diff
@@ -21,6 +21,11 @@
 public interface ConfigurationChangeNotifier {
 
     /**
+     * Begins the associated notification service provided by the given implementation.  In most implementations, no action will occur until this method is invoked.
+     */
+    void initialize();
```

Comment:
Actually, given that this is a separate task, how would you feel about initialize in its current context being a #start method and then #initialize providing the properties as discussed?